native/src/main/java/io/ballerina/stdlib/kafka/nativeimpl/consumer/Poll.java

```diff
@@ -96,7 +97,12 @@ public static Object pollPayload(Environment env, BObject consumerObject, BDecim
             try {
                 Duration duration = Duration.ofMillis(getMilliSeconds(timeout));
                 ConsumerRecords recordsRetrieved = kafkaConsumer.poll(duration);
-                ArrayType arrayType = (ArrayType) bTypedesc.getDescribingType();
+                ArrayType arrayType;
+                if (bTypedesc.getDescribingType() instanceof IntersectionType) {
+                    arrayType = (ArrayType) TypeUtils.getImpliedType(bTypedesc.getDescribingType());
+                } else {
+                    arrayType = (ArrayType) bTypedesc.getDescribingType();
+                }
```

Comment:
```suggestion
ArrayType arrayType = (ArrayType) TypeUtils.getImpliedType(bTypedesc.getDescribingType());
```

You can just do `getImpliedType` it will return the same type if it is not a intersection tyep or typeref type